### ISSUE TITLE
Java 10+ compatibility

### ISSUE THREAD

**Issue by bennyflint** [dropwizard/dropwizard#2450]
The version of Jersey (2.25.1) that DW 1.3.5 depends on is not compatible with Java 10 because that version of Jersey has a version check. I discovered this when scanning packages for classes I compiled using Java 10. Specifically, in the repackaged jersey.repackaged.org.objectweb.asm.ClassReader I find this:
```
// checks the class version
// added "support" for java 9 compiled classes
if (readShort(off + 6) > Opcodes.V9) {
    throw new IllegalArgumentException();
}
```

Unfortunately DW 1.3.5 is not compatible with the version of Jersey (2.27) that works with Java 10 because that version of Jersey does not contain AbstractValueFactoryProvider (looks like it was removed in Jersey 2.26). Here is the relevant stack trace:
```
java.lang.NoClassDefFoundError: org/glassfish/jersey/server/internal/inject/AbstractValueFactoryProvider
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1009)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:801)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:699)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:622)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
	at io.dropwizard.jersey.sessions.SessionFactoryProvider$Binder.configure(SessionFactoryProvider.java:56)
	at org.glassfish.hk2.utilities.binding.AbstractBinder.bind(AbstractBinder.java:187)
	at org.glassfish.hk2.utilities.ServiceLocatorUtilities.bind(ServiceLocatorUtilities.java:187)
	at org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.register(ImmediateHk2InjectionManager.java:88)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.glassfish.jersey.model.internal.CommonConfig.configureExternalObjects(CommonConfig.java:664)
	at org.glassfish.jersey.model.internal.CommonConfig.configureMetaProviders(CommonConfig.java:633)
	at org.glassfish.jersey.server.ResourceConfig.configureMetaProviders(ResourceConfig.java:823)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:351)
	at org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$1(ApplicationHandler.java:316)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
	at org.glassfish.jersey.internal.Errors.processWithException(Errors.java:256)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:315)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:282)
	at org.glassfish.jersey.servlet.WebComponent.<init>(WebComponent.java:335)
	at org.glassfish.jersey.servlet.ServletContainer.init(ServletContainer.java:178)
	at org.glassfish.jersey.servlet.ServletContainer.init(ServletContainer.java:370)
	at javax.servlet.GenericServlet.init(GenericServlet.java:244)
```

Seems there is a bit of a conundrum, no? Any suggestions?

-Ben

**Comment by serhiypal** [comment]
there is already commit in master that upgrades Jersey to 2.27, but wondering when it's going to be shipped

**Comment by nickbabcock** [comment]
Yes Jersey 2.27 will be included in the next release, but there is no ETA, as we need to wait for Jersey 2.28 or 2.27.1 to fix #2418

**Comment by bennyflint** [comment]
Ok, that's good to know. Thx.
